### PR TITLE
Fix Webhooks on withdraw

### PR DESCRIPTION
**Pull request by talvasconcelos**

### CODE REVIEW

**Review by dni — APPROVED** [review]

**Review by dni — APPROVED** [review]

**Comment by motorina0** [comment]
Update does not work:
 - create a link with webhook
 - update it and remove the webhook url

Error:
`TypeError: Assignment to constant variable`

image

image

File: static/js/index.js [hunk 1]
```diff
@@ -189,42 +192,28 @@ new Vue({
     },
     updateWithdrawLink: function (wallet, data) {
       var self = this
-      const body = _.pick(
```
Comment (dni) [inline static/js/index.js:192]: @motorina0 const body was removed maybe you got some old bundle files? try setting `BUNDLE_ASSETS=false`

**Review by motorina0 — APPROVED** [review]

File: static/js/index.js [hunk 2]
```diff
@@ -189,42 +192,28 @@ new Vue({
...
-          body
+          data
         )
         .then(function (response) {
           self.withdrawLinks = _.reject(self.withdrawLinks, function (obj) {
             return obj.id === data.id
           })
           self.withdrawLinks.push(mapWithdrawLink(response.data))
           self.formDialog.show = false
+          this.closeFormDialog()
```
Comment (motorina0) [inline static/js/index.js:216]: - if `.then(function (response) {` is used, then `this` refers to this function and not to the closure
- therefore `this.closeFormDialog()` will fail

Suggestion:
 - use `.then((response) => {`

File: static/js/index.js [hunk 3]
```diff
@@ -239,6 +228,7 @@ new Vue({
           self.withdrawLinks.push(mapWithdrawLink(response.data))
           self.formDialog.show = false
           self.simpleformDialog.show = false
+          this.closeFormDialog()
```
Comment (motorina0) [inline static/js/index.js:231]: see above comment https://github.com/lnbits/withdraw/pull/2/files#r1233722720

**Comment by motorina0** [comment]
LGTM. See comments.

File: static/js/index.js (hunk 2 above)
Comment (motorina0) [reply static/js/index.js:216]: fixed with new commit

Comment (talvasconcelos) [reply static/js/index.js:216]: Ahh... missed it! I usually don't use the `function` and the `self = this` thing! And missed it with `this.closeDialogForm()`

**Comment by motorina0** [comment]
@talvasconcelos are we waiting on something for this PR?

**Comment by dni** [comment]
closes #1

**Comment by talvasconcelos** [comment]
> @talvasconcelos are we waiting on something for this PR?

I don't think so... LGTM!

**Review by dni — APPROVED** [review]

**Pull request merged**